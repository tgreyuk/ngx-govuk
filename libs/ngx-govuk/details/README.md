# ngx-govuk/details

Secondary entry point of `ngx-govuk`. It can be used by importing from `ngx-govuk/details`.
