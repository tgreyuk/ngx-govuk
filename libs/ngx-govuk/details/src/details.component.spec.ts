import { Component, DebugElement } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { NgxGovukDetails } from './details.component';

@Component({
  standalone: true,
  imports: [NgxGovukDetails],
  template: `<ngx-govuk-details summaryText="Summary text">
    Summary details
  </ngx-govuk-details>`,
})
class TestHostComponent {}

describe('NgxGovukDetailsComponent', () => {
  let fixture: ComponentFixture<TestHostComponent>;
  let component: NgxGovukDetails;
  let debugElement: DebugElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [NgxGovukDetails],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(TestHostComponent);
    fixture.detectChanges();
    debugElement = fixture.debugElement.query(By.directive(NgxGovukDetails));
    component = debugElement.componentInstance;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render default state', () => {
    expect(debugElement.nativeNode).toMatchSnapshot();
  });
});
