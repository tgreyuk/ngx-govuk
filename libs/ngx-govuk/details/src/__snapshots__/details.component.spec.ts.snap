// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NgxGovukDetailsComponent should render default state 1`] = `
<ngx-govuk-details
  summarytext="Summary text"
>
  <details
    class="govuk-details"
  >
    <summary
      class="govuk-details__summary"
    >
      <span
        class="govuk-details__summary-text"
      >
        Summary text
      </span>
    </summary>
    <div
      class="govuk-details__text"
    >
       Summary details 
    </div>
  </details>
</ngx-govuk-details>
`;
